### ISSUE TITLE
Support for existing aliases

### ISSUE THREAD

**Issue by escapewindow** [epwalsh/obsidian.nvim#196]
### 🚀 The feature, motivation and pitch

Tl;dr: allow for autocompletion of existing aliases.

I'm using Obsidian as a worldbuilding/novel writing wiki, and I often have a number of terms or names for the same thing, e.g. the `Detective` page might have `aliases: [Amanda Green, Detective Green, Mandy]` in the header. This allows for:
- backlinks all link to the same page
- less duplication of content
- allows for the text referring to the page to change depending on context, using the construct `[[Detective|Detective Green]]`.

If I type `[[Amanda` in Obsidian, it will offer to autocomplete to `[[Detective|Amanda Green]]`. However, `obsidian.nvim` doesn't know about the alias, so it'll offer to create a new `Amanda` page. Furthermore, if I type `[[Detective|Amanda Green]]` manually, it'll offer to create a `[[Detective|Amanda Green|Detective|Amanda Green]]` reference.

I'd like `obsidian.nvim` to behave more like Obsidian here -- offer to autocomplete to existing aliases.

**Comment by epwalsh** [comment]
Hey @escapewindow this must be a bug or something weird going on with your setup because it _should_ work that way for you. It does for me. What happens if you type `[[Detective`? Do you get the autocomplete for Detective.md?

**Comment by escapewindow** [comment]
Yes, the filenames get autocomplete-suggested just fine. The existing aliases don't get suggested.

I did set `disable_frontmatter` since it kept changing my frontmatter in ways I didn't want it to; does that matter?

**Comment by epwalsh** [comment]
`disable_frontmatter` shouldn't affect the search results. I'm wondering if we have a YAML parsing issue. Can you post exactly what the frontmatter looks like for the note in question?

**Comment by escapewindow** [comment]
Aha, that's giving me ideas.

The frontmatter of the note in question is fairly simple and shouldn't have yaml parsing issues:

```
---
tags: [one, two]
alias: [three, four]
---
```

However, I have a python script that crawls my markdown files and parses the frontmatter, and I have to strip out the wiki links (`[[` and `]]`) before it can yaml parse many of the notes' frontmatter. I found that adding the wiki links helped in Obsidian, since it offers autocomplete, but it doesn't provide clickable links, backlinks, or the like, so I could strip all of those out if that's the culprit.

**Comment by escapewindow** [comment]
Ok, I've updated all of my frontmatter in all files, not just the ones where I can't complete aliases, so their frontmatter is yaml parsable. This meant removing tags (e.g. `#todo`) and wikilinks (e.g. `[[link]]`).

Still no alias completion from the `alias: [three, four]` list in the frontmatter. However, I can alias autocomplete from the first header in the file, even if it's not listed in the alias list.

Some questions:
- Does this sound like a bug? If I `disable_frontmatter` and don't include the first header in the alias list, should it be included as an autocomplete alias?
- I store a *lot* of extra non-obsidian yaml info in many of my headers, e.g. this template: Would this break the parsing?

```yaml
---
Title: 
Location: 
Season:
tags: [scene-reference]
Characters:
POV: 
Hook:
Scene:
- Goal: 
- Conflict: 
- Setback:
Sequel:
- Reaction:
- Dilemma:
- Decision:
Cliffhanger:
Summary:
Ideas / thoughts:
- What is the POV character trying to do in this scene? (the goal)
- Why is she trying to do it? (the motivation for that goal)
- What’s in the way of her doing it? (the conflict)
- What happens if she doesn’t do it? (the stakes)
- What goes wrong (or right)? (how the story moves forward)
- What important plot or story elements are in the scene? (what you need to remember or what affects future scenes)
---
```

**Comment by escapewindow** [comment]
To reduce variables, I've created https://github.com/escapewindow/obsidian-nvim-196 with a minimal testcase.
I added the `Detective.md` file, with only an alias. I added my `obsidian.lua` nvim config, only changing the 3 path lines.

`[[Detective` will suggest the `[[Detective]]` autocomplete; `[[Amanda` shows this:

![IMG_0078](https://github.com/epwalsh/obsidian.nvim/assets/826343/8b0c88de-c8a8-4708-ac41-95730853df0a)

**Comment by epwalsh** [comment]
Ah, okay this was definitely a YAML parsing issue. #199 should fix. Can you test that out when you get a chance?

**Comment by escapewindow** [comment]
> Ah, okay this was definitely a YAML parsing issue. #199 should fix. Can you test that out when you get a chance?

That works! \o/
Thank you!
